needy/projects/boostbuild.py

```diff
@@ -13,7 +13,20 @@ def identifier():
 
     @staticmethod
     def is_valid_project(definition, needy):
-        return os.path.isfile('Jamroot')
+        jamroot_file = None
+        for name in BoostBuildProject.__valid_jamroot_filenames():
+            if os.path.isfile(name):
+                jamroot_file = name
```

Comment:
I think this could be cleaner if you just returned here and dropped the variable.